warp10/src/main/java/io/warp10/script/WarpScriptMacroRepository.java

```diff
@@ -102,8 +104,24 @@
   /**
    * Actual macros
    */
-  private final static Map<String,Macro> macros = new HashMap<String,Macro>();
+  private final static Map<String,Macro> macros;
  
+  private static final int DEFAULT_CACHE_SIZE = 10000;
+  
+  static {
+    //
+    // Create macro map
+    //
+    
+    final int maxcachesize = Integer.parseInt(WarpConfig.getProperty(Configuration.REPOSITORY_CACHE_SIZE, Integer.toString(DEFAULT_CACHE_SIZE)));
+    
+    macros = new LinkedHashMap<String,Macro>() {
+      @Override
+      protected boolean removeEldestEntry(java.util.Map.Entry<String,Macro> eldest) {
+        return this.size() > maxcachesize;
```

Comment:
Is there a reason there is a sensision metric for the library and not for the repo?

Reply:
Same question for logging evicted macros.

Reply:
Macros which are evicted upon loading of a jar will never get re-added. In the case of the repo, when ondemand is true the macros will get loaded anyway. The ondemand param is true by default so we could log eviction only when it is false.